lms/djangoapps/support/static/support/jsx/entitlements/components/Main/Main.jsx

```diff
@@ -21,41 +22,68 @@ const Main = props => (
       isFormOpen={props.isFormOpen}
       ecommerceUrl={props.ecommerceUrl}
       openCreationForm={props.openCreationForm}
+      entitlements={props.entitlements}
     />
   </div>
 );
 
-const MainContent = (props) => {
-  if (props.isFormOpen) {
-    return <EntitlementFormContainer />;
+class MainContent extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      searchSubmitted: false,
+    };
+
+    this.handleSearchSubmit = this.handleSearchSubmit.bind(this);
+  }
+
+  handleSearchSubmit(searchSubmitted) {
+    this.setState({ searchSubmitted });
   }
-  return (
-    <div>
-      <div className="actions">
-        <SearchContainer />
-        <Button
-          className={['btn', 'btn-primary']}
-          label="Create New Entitlement"
-          onClick={props.openCreationForm}
-        />
+
+  render() {
+    if (this.props.isFormOpen) {
+      return <EntitlementFormContainer />;
+    }
+
+    return (
+      <div>
+        <div className="actions">
+          <SearchContainer handleSearchSubmit={this.handleSearchSubmit} />
+          <Button
+            className={['btn', 'btn-primary']}
+            label="Create New Entitlement"
+            onClick={this.props.openCreationForm}
+          />
+        </div>
+        {
+          this.props.entitlements.length > 0 ?
+            <EntitlementSupportTableContainer ecommerceUrl={this.props.ecommerceUrl} /> :
+              this.state.searchSubmitted ? <div>No Result Found</div> : null
+        }
       </div>
-      <EntitlementSupportTableContainer ecommerceUrl={props.ecommerceUrl} />
-    </div>
-  );
-};
+    );
+  }
+}
+
+const mapStateToProps = state => ({
+  entitlements: state.entitlements,
+});
```

Comment:
I don't think we need redux here, it's just state internal to this component that's being used isn't it?

Reply:
The server response is stored in store and In order to display message(No Record Found) there is a need to use redux so that proper message can be shown.

Reply:
It looks like you are just hard-coding ```No Result Found``` in the render function though? Where in the code is the error message being displayed?

Reply:
The reason to hard code of error message is because currently, server is returning 200 in any case whether it is provided with search query or there is no record found against a query.In addition to it, No message is shown in case of empty response.
In order to optimise it further, backend api needs to be changed accordingly and there should be a separate ticket if it needs to be done.

Reply:
So an update would need to be made here to add the error state from the redux store when that work has been done, so I think it would be cleaner to wait and add redux to this component then (otherwise it's confusing).